src/devices/vdml_usd.cpp

```diff
@@ -76,7 +89,7 @@ std::vector<std::string> list_files(const char* path) {
 		file_name = std::string_view(str.data() + index, delimiter_pos - index);
 		// This is point where content of the std::string_view file name is copied to its
 		// own std::string and added to the files vector
-		files.emplace_back(file_name);
+		files.emplace_back("/usd" + std::string(path) + "/" + std::string(file_name));
```

Comment:
You can't do it like this, this creates 7 different `std::string`s (each of which "may" allocate and pretty likely at least 2 will, wasting a lot of performance).
I don't believe the version of libstdc++ (the standard library) PROS uses supports `std::format` but if so you could write the very nice:
```cpp
files.emplace_back(std::format("/{}/{}", path, file_name));
```
if not i would recommend something more like
```cpp
const size_t path_size = 1 + usd_prefix.length() + path.length() + 1 + file_name.length() + 1; // +1's for the slashes, null terminator 
std::string buf;
buf.reserve(path_size); // Reserve enough space in the internal array that we can snprintf right into it

// The %.*s format specifier allows passing the size of a format arg before the value of the arg, so it lets
// us use the string_view into `str` with the C-style snprintf function even though it does not have the
// null terminator that would usually be required to know when a string ends.
std::snprintf(buf.data(), path_size, "/%s/%.*s", path, static_cast<int>(file_name.length()), file_name.data());

files.push_back(std::move(buf));
```
This retains the property of the old version that it only had to allocate the string once.

Reply:
> I don't believe the version of libstdc++ (the standard library) PROS uses supports std::format

Sadly it does not :(

Reply:
@Gracelu128 Ty but
1. I would recommend keeping the comments (or rewriting them in your own words if you want). Always better to explain what you're doing, especially if it may be non-obvious why (and I think readers of this code may not immediately understand what the `%.*s` does, and why its needed. I know I wouldn't know what that format specifier means at first glance)
2. Sorry, I actually made an edit to the code block after you looked at it I think:
> (EDIT: removed the + 1 in path_size calculation for null terminator, as I realized that std::string::reserve implicitly adds 1 for that by itself, and add it back in snprintf)

You could also just subtract 1 from the param to the `reserve` call if you think thats preferable.